code/modules/mob/living/carbon/xenomorph/castes/hunter/abilities_hunter.dm

```diff
@@ -439,7 +441,17 @@
 
 /datum/action/xeno_action/mirage/action_activate()
 	succeed_activate()
-	add_cooldown()
+	if (!illusions_active)
+		spawn_illusions()
+		illusions_active = TRUE
+		//Deactivates illusions_active and sets cooldown if swap not used
+		addtimer(CALLBACK(src, .proc/clean_illusions), illusion_life_time) 
+	else
+		swap()
+		add_cooldown()
```

Comment:
you should allow one swap per mirage, so you wouldn't need to use another cooldown.

Reply:
Replaced with a swap_used var which is checked in can_use_action instead of the complicated cooldown logic